libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/Tracer.java

```diff
@@ -20,36 +20,71 @@
+     * @return span, must be closed.
+     */
+    Span startSpan(SpanCreationContext context);
 
     /**
      * Starts the {@link Span} with given name
      *
      * @param spanName span name
-     * @return scope of the span, must be closed with explicit close or with try-with-resource
+     * @return span, must be closed.
      */
-    SpanScope startSpan(String spanName);
+    Span startSpan(String spanName);
 
     /**
      * Starts the {@link Span} with given name and attributes. This is required in cases when some attribute based
      * decision needs to be made before starting the span. Very useful in the case of Sampling.
-     * @param spanName span name.
+     *
+     * @param spanName   span name.
      * @param attributes attributes to be added.
-     * @return scope of the span, must be closed with explicit close or with try-with-resource
+     * @return span, must be closed.
      */
-    SpanScope startSpan(String spanName, Attributes attributes);
+    Span startSpan(String spanName, Attributes attributes);
 
     /**
      * Starts the {@link Span} with the given name, parent and attributes.
-     * @param spanName span name.
+     *
+     * @param spanName   span name.
      * @param parentSpan parent span.
      * @param attributes attributes to be added.
-     * @return scope of the span, must be closed with explicit close or with try-with-resource
+     * @return span, must be closed.
      */
-    SpanScope startSpan(String spanName, SpanContext parentSpan, Attributes attributes);
+    Span startSpan(String spanName, SpanContext parentSpan, Attributes attributes);
 
     /**
      * Returns the current span.
      * @return current wrapped span.
      */
     SpanContext getCurrentSpan();
+
+    /**
+     * Start the span and scoped it. This must be used for scenarios where {@link SpanScope} and {@link Span} lifecycles
+     * are same and ends within the same thread where created.
+     * @param spanCreationContext span creation context
+     * @return scope of the span, must be closed with explicit close or with try-with-resource
+     */
+    ScopedSpan startScopedSpan(SpanCreationContext spanCreationContext);
+
+    /**
+     * Start the span and scoped it. This must be used for scenarios where {@link SpanScope} and {@link Span} lifecycles
+     * are same and ends within the same thread where created.
+     * @param spanCreationContext span creation context
+     * @param parentSpan parent span.
+     * @return scope of the span, must be closed with explicit close or with try-with-resource
+     */
+    ScopedSpan startScopedSpan(SpanCreationContext spanCreationContext, SpanContext parentSpan);
+
+    /**
+     * Creates the Span Scope for a current thread. It's mandatory to scope the span just after creation so that it will
+     * automatically manage the attach /detach to the current thread.
+     * @param span span to be scoped
+     * @return ScopedSpan
+     */
+    SpanScope createSpanScope(Span span);
```

Comment:
May be change the name a bit for readability:

```suggestion
    SpanScope withSpanInScope(Span span);
```

To show case the example:

```
try (tracer.withSpanInScope(span)) {
  ...
}
```